jujugui/static/gui/src/app/components/terminal/terminal.js

```diff
@@ -78,17 +83,58 @@ class Terminal extends React.Component {
       }
       // Terminado sends a "disconnect" message when the process it's running
       // exits. When we receive that, we close the terminal.
-      if (resp['0'] === 'disconnect') {
-        this.close();
+      switch(resp[0]) {
+        case 'disconnect':
+          this.close();
+          break;
+        case 'setup':
+          this.terminalSetup = true;
+          break;
+        case 'stdout':
+          if (this.terminalSetup && !this.initialCommandsSent) {
+            // If the first PS1 presented to the user changes then this will
+            // need to be updated.
+            if (resp[1].indexOf('\u001b[01;32') === 0) {
```

Comment:
Let's be more defensive here, let's check that resp includes exactly two items before indexing [1]. Also, can we index on the $ part of the prompt, so that it less likely to change in the future?